jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/TopicSearchController.java

```diff
@@ -76,6 +81,8 @@ public TopicSearchController(TopicFetchService topicSearchService,
     @RequestMapping(value = "/search/index/rebuild")
     @ResponseStatus(value = HttpStatus.NO_CONTENT)
     public void rebuildIndexes() {
+        long forumComponentId = componentService.getComponentOfForum().getId();
+        componentService.checkPermissionsForComponent(forumComponentId);
```

Comment:
I think we usually put these lines into `topicSearchService.rebuildSearchIndex()`.